diff --git a/Cargo.toml b/Cargo.toml
index 6269407..db31eef 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -4,6 +4,7 @@ version = "0.3.7"
 edition = "2021"
 
 [dependencies]
+arboard = "3.3.0"
 scanf = "1.2.1"
 sha1 = "0.10.1"
 home = "0.5.9"
diff --git a/src/tui/clipboard.rs b/src/tui/clipboard.rs
index 9ad8932..a687513 100644
--- a/src/tui/clipboard.rs
+++ b/src/tui/clipboard.rs
@@ -1,3 +1,4 @@
+use arboard;
 #[derive(Debug)]
 pub struct Clipboard {
     buffer:String,
@@ -5,16 +6,28 @@ pub struct Clipboard {
 
 impl Clipboard {
     pub fn new() -> Self {
+        let clipboard = match arboard::Clipboard::new() {
+            Ok(some) => Some(some),
+            Err(_) => None,
+        };
+
         Clipboard {
             buffer:String::new(),
         }
     }
 
     pub fn get_text(&self) -> String {
-        self.buffer.clone()
+        if let Some(clipboard) = &mut self.clipboard {
+            if let Ok(text) = clipboard.get_text() {
+                return text
+            }
+        }
+        String::new()
     }
 
     pub fn set_text(&mut self, text: String) {
-        self.buffer = text;
+        if let Some(clipboard) = &mut self.clipboard {
+            let _ = clipboard.set_text(todo_string);
+        }
     }
 }
